common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -133,6 +148,14 @@ public boolean verifySignature(final String brokerPackageName) {
         return false;
     }
 
+    public Set<BrokerData> getValidBrokers() {
```

Comment:
rather than return a list of valid brokerData publicly, could we implement an isBrokerValid(pkgName, signature)?

Reply:
I've added a method for `isBrokerValid` here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1191/commits/2114def4600f39c7f09390f00bd361a76d13f941 (just in case if someone needs to use it in future)
 
but I've still kept the `getValidBrokers` method because in the places where this is being used we don't know the signature hash....we are basically just using to check if the package name that we have, matches the package name of a valid Broker app.

If we really want to NOT use `getValidBrokers` and rather use `isBrokerValid(pkgName, signature)` then we would have to calculate the signature hash (again) in these places and become aware of it and pass it along to the method.

An alternative would be to create an overload method called as `isBrokerValid(pkgName)` - this would return whether the package name maps to a valid broker app (completely ignores the signature hash aspect of broker validity)

Reply:
@AdamBJohnsonx Any thoughts on the above ^^?

Reply:
It's just that it looks like we're mainly getting broker data by package name except for the case where we compute a signature hash and then see if it was the right one for any of the brokers.  I don't want to add unused code, but every time I look at this it seems like we're repeating the same loop over the brokers set.

If nothing is using it, don't include it.